### PR TITLE
Inform of active actions when changing the session

### LINKED ISSUE #3456: Bug: Paused Scans Wiped Out when Persisting a Session
If you pause a scan, and then Save a session with 'Persist Session' when the save is done, the paused scan is 'gone'. Seems like it shouldn't have any affect on an active scan. If there is no easy way to fix this, please at least warn the user that the currently active scan is going to get wiped out so they can make the correct choice for their situation.

### PR DESCRIPTION
**Pull request by thc202**
Change MenuFileControl to inform of active actions (for example, spider,
active scans) before creating, opening, or saving the session.

Fix #3456 - Paused Scans Wiped Out when Persisting a Session

### CODE REVIEW

File: src/org/parosproxy/paros/control/MenuFileControl.java [hunk 1]
```diff
@@ -164,6 +169,32 @@ public void newSession(boolean isPromptNewSession) throws ClassNotFoundException
 				break;
 		}
 	}
+
+	private boolean informStopActiveActions() {
+		String activeActions = wrapEntriesInLiTags(control.getExtensionLoader().getActiveActions());
+		if (!activeActions.isEmpty()) {
+			String message = MessageFormat
```
Comment (kingthorin) [inline src/org/parosproxy/paros/control/MenuFileControl.java]: Couldn't this just be: `Constant.messages.getString("menu.file.session.activeactions", activeActions)` ?

File: src/org/parosproxy/paros/control/MenuFileControl.java [hunk 2]
```diff
@@ -164,6 +169,32 @@ public void newSession(boolean isPromptNewSession) throws ClassNotFoundException
 				break;
 		}
 	}
+
+	private boolean informStopActiveActions() {
+		String activeActions = wrapEntriesInLiTags(control.getExtensionLoader().getActiveActions());
+		if (!activeActions.isEmpty()) {
+			String message = MessageFormat
+					.format(Constant.messages.getString("menu.file.session.activeactions"), activeActions);
```
Comment (kingthorin) [inline src/org/parosproxy/paros/control/MenuFileControl.java]: Couldn't this just be: `Constant.messages.getString("menu.file.session.activeactions", activeActions)` ? (Without the MessageFormat)

File: src/org/parosproxy/paros/control/MenuFileControl.java [hunk 3]
```diff
@@ -164,6 +169,32 @@ public void newSession(boolean isPromptNewSession) throws ClassNotFoundException
...
+			if (view.showConfirmDialog(message) != JOptionPane.OK_OPTION) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	private static String wrapEntriesInLiTags(List<String> entries) {
+		if (entries.isEmpty()) {
+			return "";
```
Comment (kingthorin) [inline src/org/parosproxy/paros/control/MenuFileControl.java]: Should we log something here? If the "actions" are empty wouldn't that be something strange?

File: src/org/parosproxy/paros/control/MenuFileControl.java (hunk 2 above)
Comment (thc202) [inline src/org/parosproxy/paros/control/MenuFileControl.java]: Sure.

File: src/org/parosproxy/paros/control/MenuFileControl.java (hunk 3 above)
Comment (thc202) [inline src/org/parosproxy/paros/control/MenuFileControl.java]: No, that's expected to happen, when nothing is running/active.

File: src/org/parosproxy/paros/control/MenuFileControl.java (hunk 2 above)
Comment (thc202) [inline src/org/parosproxy/paros/control/MenuFileControl.java]: Change done.

File: src/org/parosproxy/paros/control/MenuFileControl.java (hunk 3 above)
Comment (kingthorin) [inline src/org/parosproxy/paros/control/MenuFileControl.java]: Ok I was thinking more of some sort of weird race or time of check/time of use issue where we check something is running but then by the time it's added to the list it's done....

All good.

Comment (thc202) [inline src/org/parosproxy/paros/control/MenuFileControl.java]: That case can happen, but no very problematic IMO. Warrants further changes?

Comment (kingthorin) [inline src/org/parosproxy/paros/control/MenuFileControl.java]: No, probably fine. If it comes up we've thought about it and can tackle it if seen.

Comment (thc202) [inline src/org/parosproxy/paros/control/MenuFileControl.java]: OK.

**Pull request merged**